### PR TITLE
GAWB-3052: Convert AgoraService.queryRoute (skip tests)

### PR DESCRIPTION
**Pull request by kyuksel**
Left in a couple of commented out blocks as they have stuff that makes it easier for me to convert the remaining routes. Will remove them once all related routes are converted.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala [hunk 1]
```diff
@@ -19,60 +19,58 @@ import scala.concurrent.{ExecutionContext, Future}
  * It then handles the returns from the business layer and completes the request. It is responsible for querying the
  * methods repository for methods and method configurations.
  */
-class QueryHandler(dataSource: PermissionsDataSource, implicit val ec: ExecutionContext) extends Actor {
-  implicit val system = context.system
-
+class QueryHandler(dataSource: PermissionsDataSource)(implicit ec: ExecutionContext) {
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala:22]: I think this class should go away. The usage of it is a direct passthrough to`agoraBusiness.find`

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/routes/RouteHelpers.scala [hunk 2]
```diff
@@ -239,11 +237,11 @@ trait BaseRoute extends RouteUtil  {
 //    val message = QueryCompatibleConfigurations(context, namespace, name, snapshotId, username)
 //    perRequest(context, queryHandler, message)
 //  }
-//}
+}
 
 trait AddRouteHelper extends BaseRoute {
   def postPath(_path: String)(implicit ec: ExecutionContext) = {
-    post & versionedPath(_path) & authenticationDirectives.usernameFromRequest(())
+    post & versionedPath(_path) & authenticationDirectives.usernameFromRequest()
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/routes/RouteHelpers.scala:244]: nice

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/ServiceMessages.scala [hunk 3]
```diff
@@ -89,11 +89,11 @@ object ServiceMessages {
...
-  case class Query(requestContext: RequestContext,
-                   agoraSearch: AgoraEntity,
-                   agoraProjection: Option[AgoraEntityProjection],
-                   entityTypes: Seq[AgoraEntityType.EntityType],
-                   username: String)
+//  case class Query(requestContext: RequestContext,
+//                   agoraSearch: AgoraEntity,
+//                   agoraProjection: Option[AgoraEntityProjection],
+//                   entityTypes: Seq[AgoraEntityType.EntityType],
+//                   username: String)
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/ServiceMessages.scala:96]: This class will need a lot of cleanup ... tons of stuff never used.

**Review by rushtong — APPROVED** [review]
👍 Looks good! One minor suggestion that doesn't need to be acted upon yet unless you want to.

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala (hunk 1 above)
Comment (kyuksel) [reply src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala:22]: Agreed. Will take care of it as part of the next PR.

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/ServiceMessages.scala (hunk 3 above)
Comment (kyuksel) [reply src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/ServiceMessages.scala:96]: Agreed. It looks like we'll be able to remove it altogether.

**Pull request merged**